src/ATen/native/xpu/sycl/Atomics.h

```diff
@@ -115,6 +115,93 @@ struct AtomicIntegerImpl<T, 8> {
         address, val, [](DTYPE a, DTYPE b) { return OP; }); \
   }
 
+template <typename T, size_t n>
+struct AtomicIntegerImplLocal;
+
+template <typename T>
+struct AtomicIntegerImplLocal<T, 1> {
```

Comment:
What's the operation is done with atomic here?
Why don't we use `AtomicRef<int32>` directly, but use CAS here?

Reply:
use fetch_max instead, done